src/main/java/club/minnced/discord/webhook/send/component/ActionComponent.java

```diff
@@ -0,0 +1,16 @@
+package club.minnced.discord.webhook.send.component;
+
+import org.jetbrains.annotations.Nullable;
+
+/**
+ * Components that can be inserted inside layout components (buttons & select menus)
+ */
+public interface ActionComponent extends Component {
```

Comment:
Add `withDisabled` and `isDisabled` to this, with overrides on button and select menu to return their respective types.